### DISCUSSION TITLE
Click event on minimap

### DISCUSSION THREAD

**Discussion by benbaja** [katspaugh/wavesurfer.js#3836] [Q&A] [answered]
Hi!
I'm looking for a way to subscribe the minimap to click or drag events in order to add a zoom functionality (i.e. the Ableton Live minimap in arrangement view), is there a simple way to do this that I missed?
Thanks a lot!

**Comment by katspaugh** [comment, marked as answer]
Minimap only exposes an “interaction” event atm (it’s just click basically). I think it would make sense to emit all (most?) of wavesurfer events, so pull requests are welcome.

**Reply by benbaja** [reply]
Sounds good! I’ll shoot my shot at a pull request then 😄

**Reply by benbaja** [reply]
Would adding a "getWrapper()" function similar to a wavesurfer instance be a welcome pull request as well? I realized my idea wouldn't work without adding an event listener to the wrapper itself in order to get the Y scrolling coordinate...

**Reply by katspaugh** [reply]
Ideally you could subscribe to drag events using an internal reference to the wrapper and emit (public) zoom events. You can use the [makeDraggable](https://github.com/katspaugh/wavesurfer.js/blob/main/src/draggable.ts) function for that.
